packages/frontend-v2/components/Header/UserDropdown.tsx

```diff
@@ -0,0 +1,83 @@
+import {
+  Menu,
+  MenuButton,
+  MenuList,
+  MenuItem,
+  Link,
+  Icon,
+  IconProps,
+  Text,
+} from "@chakra-ui/react";
+import React from "react";
+import { useStudentWithPlans } from "../../hooks";
+import { handleApiClientError, logger, logout} from "../../utils";
+import { useRouter } from "next/router";
+import { LoadingPage } from "../Spinner";
+import { AccountOverview } from "./AccountOverview";
+import { ChangePassword } from "./ChangePassword";
+
+export const UserDropdown: React.FC = () => {
+  const { error, student, mutateStudent } = useStudentWithPlans();
+  const router = useRouter();
+  
+  if (error && !student) {
+    logger.error("HomePage", error);
+    handleApiClientError(error, router);
+
+    // If we couldn't fetch the student, show a blank page for now.
+    // We might want to show some more actionable error in the future.
+    return <div></div>;
+  }
+
+  if (!student) {
+    return <LoadingPage />;
```

Comment:
LoadingPage is for Page component top level return only... need to use the Spinner component used within the LoadingPage component. Yall can test this out by just returning the Spinner even when the student exist to ensure the loading state looks fine. Add the screen shot to the description when done!